backends/tofino/bf-asm/salu_inst.cpp

```diff
@@ -550,8 +550,9 @@ Instruction *AluOP::pass1(Table *tbl_, Table::Actions::Action *act) {
     if (!k1) k1 = k2;
     if (k1 && (k1->value < Target::STATEFUL_ALU_CONST_MIN() ||
                k1->value > Target::STATEFUL_ALU_CONST_MAX())) {
-        if (k1->value >= (INT64_C(1) << tbl->alu_size()) ||
-            k1->value < (~UINT64_C(0) << (tbl->alu_size() - 1))) {
+        auto min_value = -(INT64_C(1) << (tbl->alu_size() - 1));
+        auto max_value = (INT64_C(1) << tbl->alu_size()) - 1;
+        if (k1->value < min_value || k1->value >= max_value) {
```

Comment:
With this max value (extra -1), it should be `if (k1->value < min_value || k1->value > max_value)`